snuba/replacer.py

```diff
@@ -29,6 +29,12 @@ def __init__(
         ), f"This storage writer does not support replacements {type(storage)}"
         self.__replacer_processor = processor
 
+        # TODO: this is a leaky abstraction, but should serve us for the patchfile
+        self.__replacer_processor.__optimize = optimize
```

Comment:
Does that work? Doesn't double underscore mangle the name?

Reply:
I would advise we control the optimize flag through settings instead of a CLI parameter.
If we load it from settings we do not need the leaky abstraction to change the state of the ReplacerProcessor from outside.
We also set fewer expectations that this would be supported in the long term
We are also in general trying to move away from customizing anything related with the storage and the dataset through CLI parameter and rely on settings instead.